src/query/dblink_scan.c

```diff
@@ -161,40 +215,88 @@ dblink_make_date_time_tz (T_CCI_U_TYPE utype, DB_VALUE * value_p, T_CCI_DATE_TZ
+			       NULL);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_make_timestamptz (value_p, &tz_timestamp);
       break;
     case CCI_U_TYPE_DATETIMETZ:
-      db_datetime_encode (&t_datetime, date_time_tz->mon, date_time_tz->day, date_time_tz->yr,
-			  date_time_tz->hh, date_time_tz->mm, date_time_tz->ss, date_time_tz->ms);
-      tz_create_datetimetz (&t_datetime, date_time_tz->tz, strlen (date_time_tz->tz), &region, &tz_datetime, NULL);
-      db_make_datetimetz (value_p, &tz_datetime);
+      error = db_datetime_encode (&t_datetime, date_time_tz->mon, date_time_tz->day, date_time_tz->yr,
+				  date_time_tz->hh, date_time_tz->mm, date_time_tz->ss, date_time_tz->ms);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error =
+	tz_create_datetimetz (&t_datetime, date_time_tz->tz, strlen (date_time_tz->tz), &region, &tz_datetime, NULL);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_make_datetimetz (value_p, &tz_datetime);
       break;
     case CCI_U_TYPE_TIMESTAMPLTZ:
-      db_time_encode (&t_time, date_time_tz->hh, date_time_tz->mm, date_time_tz->ss);
-      db_date_encode (&t_date, date_time_tz->mon, date_time_tz->day, date_time_tz->yr);
-      tz_create_timestamptz (&t_date, &t_time, date_time_tz->tz, strlen (date_time_tz->tz), &region, &tz_timestamp,
-			     NULL);
-      db_make_timestampltz (value_p, tz_timestamp.timestamp);
+      error = db_time_encode (&t_time, date_time_tz->hh, date_time_tz->mm, date_time_tz->ss);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_date_encode (&t_date, date_time_tz->mon, date_time_tz->day, date_time_tz->yr);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error =
+	tz_create_timestamptz (&t_date, &t_time, date_time_tz->tz, strlen (date_time_tz->tz), &region, &tz_timestamp,
+			       NULL);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_make_timestampltz (value_p, tz_timestamp.timestamp);
       break;
     case CCI_U_TYPE_DATETIMELTZ:
-      db_datetime_encode (&t_datetime, date_time_tz->mon, date_time_tz->day, date_time_tz->yr,
-			  date_time_tz->hh, date_time_tz->mm, date_time_tz->ss, date_time_tz->ms);
-      tz_create_datetimetz (&t_datetime, date_time_tz->tz, strlen (date_time_tz->tz), &region, &tz_datetime, NULL);
-      db_make_datetimeltz (value_p, &tz_datetime.datetime);
+      error = db_datetime_encode (&t_datetime, date_time_tz->mon, date_time_tz->day, date_time_tz->yr,
+				  date_time_tz->hh, date_time_tz->mm, date_time_tz->ss, date_time_tz->ms);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error =
+	tz_create_datetimetz (&t_datetime, date_time_tz->tz, strlen (date_time_tz->tz), &region, &tz_datetime, NULL);
+      if (error != NO_ERROR)
+	{
+	  break;
+	}
+      error = db_make_datetimeltz (value_p, &tz_datetime.datetime);
       break;
     default:
+      assert (false);
```

Comment:
```suggestion
      error = ER_FAILED; // proper error, I just set dummy one for code review.
      assert (false);
```

Reply:
it must not be reached because this routine is called with only the proper date/time zone type from dblink_scan_next.
it should return an error, we need a proper error number.